src/components/organisms/AdminVenueView/components/VenueRoomsEditor/VenueRoomsEditor.tsx

```diff
@@ -0,0 +1,298 @@
+import React, {
+  useCallback,
+  useState,
+  useMemo,
+  CSSProperties,
+  Dispatch,
+  SetStateAction,
+  useEffect,
+} from "react";
+import { useDrop } from "react-dnd";
+import update from "immutability-helper";
+import ReactResizeDetector from "react-resize-detector";
+import classNames from "classnames";
+
+import { Dimensions, Position } from "types/utility";
+import { RoomData_v2 } from "types/rooms";
+
+import { DragItem } from "pages/Account/Venue/VenueMapEdition/interfaces";
+import { ItemTypes } from "pages/Account/Venue/VenueMapEdition/ItemTypes";
+import { DraggableSubvenue } from "pages/Account/Venue/VenueMapEdition/DraggableSubvenue";
+import { CustomDragLayer } from "pages/Account/Venue/VenueMapEdition";
+import { snapToGrid as doSnapToGrid } from "pages/Account/Venue/VenueMapEdition/snapToGrid";
+
+import "./VenueRoomsEditor.scss";
+
+const styles: React.CSSProperties = {
+  width: "100%",
+  position: "relative",
+};
+
+export interface RoomIcon {
+  top: number;
+  left: number;
+  url: string;
+  width: number;
+  height: number;
+  roomIndex?: number;
+}
+
+export interface RoomIconsMap {
+  [key: string]: RoomIcon;
+}
+
+interface PortalSize {
+  width: number;
+  height: number;
+}
+
+export interface VenueRoomsEditorProps {
+  snapToGrid?: boolean;
+  roomIcons?: RoomIcon[];
+  backgroundImage: string;
+  iconImageStyle?: CSSProperties; // This is not being used ATM
+  draggableIconImageStyle?: CSSProperties; // This is not being used ATM
+  roomIconsMap?: RoomIconsMap;
+  onOtherIconClick?: (key: string) => void;
+  coordinatesBoundary: PortalSize;
+  interactive: boolean;
+  resizable: boolean;
+  onResize?: (size: Dimensions) => void;
+  onMove?: (position: Position) => void;
+  otherIconsStyle?: CSSProperties;
+  rounded?: boolean;
+  backgroundImageStyle?: CSSProperties;
+  containerStyle?: CSSProperties;
+  lockAspectRatio?: boolean;
+  rooms: RoomData_v2[];
+  selectedRoom: RoomData_v2 | undefined;
+  setSelectedRoom: Dispatch<SetStateAction<RoomData_v2 | undefined>>;
+}
+
+export const VenueRoomsEditor: React.FC<VenueRoomsEditorProps> = ({
+  snapToGrid,
+  roomIcons,
+  backgroundImage,
+  iconImageStyle,
+  coordinatesBoundary,
+  interactive,
+  resizable,
+  rounded,
+  backgroundImageStyle,
+  containerStyle,
+  lockAspectRatio,
+  rooms,
+  selectedRoom,
+  setSelectedRoom,
+  onResize,
+  onMove,
+}) => {
+  const roomIconsArray = useMemo(() => roomIcons ?? [], [roomIcons]);
+
+  const [boxes, setBoxes] = useState<RoomIconsMap>({});
+  const [imageDims, setImageDims] = useState<Dimensions>();
+
+  const convertDisplayedCoordToIntrinsic = useCallback(
+    (
+      val: number,
+      dimension: "height" | "width",
+      coordinateBoundary: number
+    ) => {
+      if (!imageDims) return 0;
```

Comment:
```suggestion
      if (!imageDims) return 0;
      
```

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1583/commits/e3b301bfa611c387d3e2ac24b7195994415ffc70